crates/electrum/src/electrum_ext.rs

```diff
@@ -165,20 +165,15 @@ pub trait ElectrumExt<A> {
     /// The scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
     /// transactions. `batch_size` specifies the max number of script pubkeys to request for in a
     /// single batch request.
-    fn scan<K: Ord + Clone>(
+    fn sync_with_keychain<K: Ord + Clone>(
         &self,
-        prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
-        stop_gap: usize,
-        batch_size: usize,
```

Comment:
You will still need the `prev_tip`, `stop_grap` and `batch_size`. `stop_gap` tells you to stop searching a keychain for outputs associated to its scripts if you haven't found any outputs after `stop_gap` derivation indices.